### PR TITLE
Added copy to clipboard functionality

### LINKED ISSUE #143: Add "Copy" button in problem page to copy sample input and sample output to the clipboard
In the route : http://localhost:3000/problem/:qNo
Add a copy button with the sample input and the sample output sections, add a copy button, which would copy the corresponding text to the clipboard.

### PR DESCRIPTION
**Pull request by Piyush3436**
Fixed #143 
# In the route : http://localhost:3000/problem/:qId
# Added a copy button with the sample input and the sample output sections, which copies the corresponding text to the clipboard.

# The above functionality was added by defining a new function "copyToClipboard". When clicked on the button, text is copied to the clipboard and an alert box appears displaying the message. 

# 
image

### CODE REVIEW

**Review by Signior-X — CHANGES_REQUESTED** [review]

**Comment by Signior-X** [comment]
@Piyush3436 are you working on this?

**Comment by Piyush3436** [comment]
> @Piyush3436 are you working on this?

@Signior-X changes done

File: views/problem_display.ejs [hunk 1]
```diff
@@ -53,7 +54,22 @@
     <footer>
         <% include partials/footer %>
     </footer>
-    
+    <script>
+    function copyToClipboard(id) {
+    const string = id.innerHTML;
```
Comment (Signior-X) [inline views/problem_display.ejs:59]: Check below, this should be str instead of string.

You should run and test and then make a commit.

File: views/problem_display.ejs [hunk 2]
```diff
@@ -37,11 +37,13 @@
...
-                <pre><%= data.constraints %></pre>
+                <pre><%= data.constraints %></pre> 
                 <h2>Sample Input</h2>
-                <pre><%= data.sampleInput %></pre>
+                <pre id = "one"><%= data.sampleInput %></pre>
+                <button class="btn btn-light btn-sm" onclick="copyToClipboard(one)">Copy text to clipboard</button>
                 <h2>Sample Output</h2>
-                <pre><%= data.sampleOutput %></pre>
+                <pre id = "two"><%= data.sampleOutput %></pre>
+                <button class="btn btn-light btn-sm" onclick = copyToClipboard(two) %>>Copy text to clipboard</button>
```
[the remark this answers is not in the document]
Comment (Signior-X) [reply views/problem_display.ejs:45]: Great,
give some padding on the left of the button so that it looks nice.

**Review by Signior-X — CHANGES_REQUESTED** [review]
Your code isn't working.
Test run it @Piyush3436

**Comment by Piyush3436** [comment]
@Signior-X I've added padding from all directions as well as margin from the left. Border has been added for the button to look better. 
The code is working too.

File: views/problem_display.ejs [hunk 3]
```diff
@@ -37,11 +37,11 @@
...
                 <h2>Constraints</h2>
-                <pre><%= data.constraints %></pre>
-                <h2>Sample Input</h2>
-                <pre><%= data.sampleInput %></pre>
-                <h2>Sample Output</h2>
-                <pre><%= data.sampleOutput %></pre>
+                <pre><%= data.constraints %></pre> 
+                <h2>Sample Input<span><button class="border border-dark ml-4 p-2 btn btn-light btn-sm" onclick="copyToClipboard(one)"><b>Copy text to clipboard</b></button></span></h2>
+                <pre id = "one"><%= data.sampleInput %></pre>
+                <h2>Sample Output<span><button class="border border-dark ml-4 p-2 btn btn-light btn-sm" onclick="copyToClipboard(one)"><b>Copy text to clipboard</b></h2>
```
Comment (Signior-X) [inline views/problem_display.ejs:43]: don't you think it should be `copyToClipboard(two)` instead of `copyToClipboard(one)`

Also it will be better if you write the id of these pre tags as something better like _sample-input-copy_ and _sample-output-copy_

**Review by Signior-X — CHANGES_REQUESTED** [review]

**Comment by Piyush3436** [comment]
@Signior-X required changes done

**Comment by Signior-X** [comment]
Sorry for the delay.. was busy somewhere.
@Piyush3436 what is this? You have mixed up some commits or branches.
See the file changes many of them are not needed in this PR.
Only for the code changes for copy to clipboard should be in this PR.

File: controls/auth.js [hunk 4]
```diff
@@ -167,7 +167,10 @@ exports.submission_subID = async (req, res, next) => {
  * route: /user/updateProfile
  */
 exports.getUpdateProfile = (req, res, next) => {
-    res.render("edit_profile", { message: null });
+    user.findById(req.params.id, (err, data) => {
```
Comment (Signior-X) [inline controls/auth.js:170]: Revert the changes of df19020

**Review by Signior-X — CHANGES_REQUESTED** [review]

**Comment by Signior-X** [comment]
@Piyush3436 after marking changes, comment or again ask for review so that the reviewer get to know that he needs to review a PR.

Also great the changes you did, however few minor changes were needed. Just one suggestion... while making PR or  making changes at least test the code properly before committing the code.
 Rest nice work done, Keep doing all the best!
